api-examples/rbacv1/main.py

```diff
@@ -0,0 +1,324 @@
+            if tm.team_name == item:
+                return True
+
+    def append(self, team):
+        self.teams.append(team)
+
+
+class SLAPIClient:
+    """
+    SLAPIClient handles communications with ShiftLeft API v4 for the purposes of this script.
+    It is very limited and bound to be obsoleted of Schema changes.
+    """
+
+    def __init__(self, access_token="", organization_id=""):
+        self.__access_header = {'Authorization': 'Bearer {}'.format(access_token)}
+        self.__organization_id = organization_id
+
+    def _do_get(self, api_path):
+        u = API_V4_BASE_URL + API_V4_ORG_PATH.format(organization_id=self.__organization_id) + api_path
+        resp = requests.get(u, self.__access_header)
+        handle_status_code(resp)
+        return resp.json().get('response', None)
+
+    def _do_post(self, api_path, payload=None):
+        u = API_V4_BASE_URL + api_path
+        resp = requests.post(u, self.__access_header, payload)
+        handle_status_code(resp)
+        return resp.json().get('response', None)
+
+    def _do_put(self, api_path, payload=None):
+        u = API_V4_BASE_URL + api_path
+        resp = requests.put(u, self.__access_header, payload)
+        handle_status_code(resp)
+        return resp.json().get('response', None)
+
+    def list_users(self):
+        """
+        list_users implements a GET request to https://docs.shiftleft.io/api/#operation/ListOrgRBACUsers
+        :return:
+        """
+        return SLListUsersResponse(self._do_get("rbac/users"))
+
+    def list_teams(self):
+        """
+        list_teams implements a GET request to https://docs.shiftleft.io/api/#operation/ListTeams
+        :return:
+        """
+        resp = self._do_get("rbac/teams")
+        return SLTeams(resp)
+
+    def list_roles(self):
+        """
+        list_roles implements a GET request to https://docs.shiftleft.io/api/#operation/ListTeams
+        :return:
+        """
+        return self._do_get("rbac/roles")
+
+    def create_team(self, name=""):
+        """
+        create_team implements a POST request to https://docs.shiftleft.io/api/#operation/CreateTeam
+        :param name: the name of the team to be created, must be unique
+        :return:
+        """
+        team_payload = {
+            "name": name
+        }
+        resp = self._do_post("rbac/teams", team_payload)
+        return SLTeamInfo(team_id=resp["team_id"], team_name=name)
+
+    def assign_user_organization_role(self, user_id="", role=""):
+        """
+        assign_user_organization_role will assign the role passed to the user at an organization level
+
+        :param user_id: the id v2 of the user
+        :param role: the role id or alias the user will have at an organization level
+        :return: a dictionary of the json response from the call.
+        """
+        user_org_role_payload = {"org_role": role}
+        self._do_put("rbac/users/{user_id}".format(user_id=user_id), user_org_role_payload)
```

Comment:
what's the minimum supported python version? because it looks like you're wanting to use f-strings here :-)

Reply:
I would tend to agree that an f string would be clearer here but I would like to keep this example valid in a few more python versions.